include/boost/math/statistics/univariate_statistics.hpp

```diff
@@ -516,10 +516,15 @@ template<class RandomAccessIterator>
 auto mode(RandomAccessIterator first, RandomAccessIterator last) -> std::vector<typename std::iterator_traits<RandomAccessIterator>::value_type>
 {
     using Z = typename std::iterator_traits<RandomAccessIterator>::value_type;
+    static_assert(std::is_integral<Z>::value, "Floating point values have not yet been implemented.");
     using Size = typename std::iterator_traits<RandomAccessIterator>::difference_type;
 
     std::vector<Z> sorted(first, last);
-    std::sort(sorted.begin(), sorted.end());
+    
+    if(!std::is_sorted(sorted.begin(), sorted.end()))
```

Comment:
I'm not convinced this is buying us much. `std::is_sorted` takes O(N) time if the vector is sorted, and `std::sort` also takes O(N) time if the vector is sorted.

I think having a `sorted_mode` would also help us with the benchmarks, because your results are currently dominated by the time for sorting.

Reply:
With a separate sorted mode function the results validate the addition.

Worst values
test_mode: Unchanged
sequential: 2.48 -> 2.01 nlog(n)
sequential_pairs: 1.87 -> 1.40 nlog(n)
sequential_multiples: 1.55 -> 1.08 nlog(n)

Reply:
Wait, shouldn't this be O(N) for the sorted input?